.github/workflows/e2e.yml

```diff
@@ -91,12 +90,28 @@ jobs:
 
       # Checkouts
 
-      - name: Checkout (frontend)
+      - name: Checkout (Push)
+        if: github.event_name == 'push'
         uses: actions/checkout@v2
 
-      - name: Set REF in env, removing the `refs/` part
+      - name: Checkout (Pull Request Target)
+        if: github.event_name == 'pull_request_target'
+        uses: actions/checkout@v2
+        with:
+          # We are in a pull_request_target event - we need to explicitly checkout the PR's head.
+          # Otherwise we would be running against main branch
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Why are we using `sha` and not `ref` if ref is reliable and we're using it later?

Reply:
@znarf : The reason we use the `ref` for previous section is because it's strictly needed there. After all we need the `ref` of the branches to match; she `sha` will not match as they are two different repos. Here, it doesn't matter whether we use `sha` or `ref` I think as it just points to the head of the branch that the pull request was made.

Reply:
@znarf : We could change it to `ref` just to be consistent? But shall we just test this out first before making this change that the `ref` in other places works as intended? 😄

Reply:
I prefer consistency, if `ref` can't be trusted then matching branches will not work and that is a blocker anyway.

Reply:
@znarf : Sure, I've changed that on both front and api. 😄